File: one-mobile/flutter_project/lib/xrauth/model/guest.dart
import 'dart:convert';

class Guest {
  String email;
  String password;
  String username;
  String nickname;
  String userId;

  Guest({
    required this.email,
    required this.password,
    required this.username,
    required this.nickname,
    required this.userId,
  });

  static Guest parse(String str) => Guest.fromJson(json.decode(str));

  static String toJsonString(Guest data) => json.encode(data.toJson());

  factory Guest.fromJson(Map<String, dynamic> json) => Guest(
        email: json['email'],
        password: json['password'],
        username: json['username'],
        nickname: json['nickname'],
        userId: json['user_id'],
      );

  Map<String, dynamic> toJson() => {
        'email': email,
        'password': password,
        'username': username,
        'nickname': nickname,
        'user_id': userId,
      };
}

File: one-unity/core/development/common/game-account/Runtime/Scripts/Model/Guest.cs
namespace TPFive.Game.Account
{
    using System;
    using Newtonsoft.Json;

    [Serializable]
    public class Guest
    {
        [JsonProperty("email")]
        public string Email { get; set; }

        [JsonProperty("password")]
        public string Password { get; set; }

        [JsonProperty("username")]
        public string Username { get; set; }

        [JsonProperty("nickname")]
        public string Nickname { get; set; }

        [JsonProperty("user_id")]
        public string UserId { get; set; }

        public static Guest Parse(string guestJson)
        {
            return JsonConvert.DeserializeObject<Guest>(guestJson);
        }

        public string ToJsonString()
        {
            return JsonConvert.SerializeObject(this);
        }
    }
}